src/main/java/com/conveyal/datatools/manager/auth/Auth0Connection.java

```diff
@@ -60,17 +77,64 @@ public static void checkUser(Request req) {
         // for downstream controllers to check permissions.
         try {
-            byte[] decodedSecret = new Base64().decode(getConfigPropertyAsText("AUTH0_SECRET"));
-            JWTVerifier verifier = new JWTVerifier(decodedSecret);
-            Map<String, Object> jwt = verifier.verify(token);
+            Map<String, Object> jwt = verifyToken(token);
+            remapTokenValues(jwt);
             Auth0UserProfile profile = MAPPER.convertValue(jwt, Auth0UserProfile.class);
+            // The user attribute is used on the server side to check user permissions and does not have all of the
+            // fields that the raw Auth0 profile string does.
             req.attribute("user", profile);
+            // The raw_user attribute is used to return the complete user profile string to the client upon request in
+            // the UserController. Previously the client sought the profile directly from Auth0 but the need to add
+            req.attribute("raw_user", MAPPER.writeValueAsString(jwt));
         } catch (Exception e) {
             LOG.warn("Login failed to verify with our authorization provider.", e);
             haltWithMessage(401, "Could not verify user's token");
         }
     }
 
+    /**
+     * Choose the correct JWT verification algorithm (based on the values present in env.yml config) and verify the JWT
+     * token.
+     */
+    private static Map<String, Object> verifyToken(String token) throws NoSuchAlgorithmException, NoSuchProviderException, InvalidKeySpecException, IOException, JWTVerifyException, InvalidKeyException, SignatureException {
+        if (verifier == null) {
+            if (hasConfigProperty("AUTH0_SECRET")) {
+                // Use HS256 algorithm to verify token (uses client secret).
+                byte[] decodedSecret = new org.apache.commons.codec.binary.Base64().decode(getConfigPropertyAsText("AUTH0_SECRET"));
+                verifier = new JWTVerifier(decodedSecret);
+            } else if (hasConfigProperty("AUTH0_PUBLIC_KEY")) {
+                // Use RS256 algorithm to verify token (uses public key/.pem file).
+                PublicKey publicKey = PemReader.readPublicKey(getConfigPropertyAsText("AUTH0_PUBLIC_KEY"));
+                verifier = new JWTVerifier(publicKey);
+            } else {
+                throw new RuntimeException("Server authentication provider not configured correctly.");
+            }
+        }
+        return verifier.verify(token);
+    }
+
+    /**
+     * Handle mapping token values to the expected keys. This accounts for app_metadata and user_metadata that have been
+     * scoped to conform with OIDC (i.e., how newer Auth0 accounts structure the user profile) as well as the user_id ->
+     * sub mapping.
+     */
+    private static void remapTokenValues(Map<String, Object> jwt) {
+        // If token did not contain app_metadata or user_metadata, add the scoped values to the decoded token object.
```

Comment:
Since this method is only used once, I think it's appropriate to lump it into the verifyToken method.